src/components/search/index.tsx

```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react';
+import { SearchProps } from './types';
+
+export default function Search({ initialValue, onSearch } :SearchProps) {
+    const [searchValue, setSearchValue] = useState(initialValue);
+
+    return (
+        <div>
+            <input
+                className='searchInput'
+                value={searchValue}
+                onChange={e => {setSearchValue(e.target.value)}}
```

Comment:
You also need to handle Enter key button

Reply:
Fixed